pkg/plugins/models/tsdb.go

```diff
@@ -0,0 +1,194 @@
+package models
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+	"github.com/grafana/grafana/pkg/components/null"
+	"github.com/grafana/grafana/pkg/components/simplejson"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/timberio/go-datemath"
+)
+
+// TSDBSubQuery represents a TSDB sub-query.
+type TSDBSubQuery struct {
+	RefID         string             `json:"refId"`
+	Model         *simplejson.Json   `json:"model,omitempty"`
+	DataSource    *models.DataSource `json:"datasource"`
+	MaxDataPoints int64              `json:"maxDataPoints"`
+	IntervalMS    int64              `json:"intervalMs"`
+	QueryType     string             `json:"queryType"`
+}
+
+// TSDBQuery contains all information about a TSDB query request.
+type TSDBQuery struct {
```

Comment:
Would try to remove all the TSDB prefixes from these types and replace them with Data

Reply:
Done.